Gems/Atom/RPI/Code/Source/RPI.Public/WindowContext.cpp

```diff
@@ -216,10 +216,10 @@ namespace AZ
                     xrDescriptor.m_dimensions.m_imageCount = AZ::RHI::Limits::Device::FrameCountMax;
                     xrDescriptor.m_isXrSwapChain = true;
                     xrDescriptor.m_xrSwapChainIndex = i;
-                    xrDescriptor.m_dimensions.m_imageFormat = GetSwapChainFormat(device);
+                    xrDescriptor.m_dimensions.m_imageFormat = xrSystem->GetSwapChainFormat(i);
 
-                    attachmentName = AZStd::string::format("XRSwapChain_View_%i", i);
-                    xrDescriptor.m_attachmentId = RHI::AttachmentId{ attachmentName.c_str() };
+                    const AZStd::string xrAttachmentName = AZStd::string::format("XRSwapChain_View_%i", i);
+                    xrDescriptor.m_attachmentId = RHI::AttachmentId{ xrAttachmentName.c_str() };
```

Comment:
This can format to an `AZStd::fixed_string` here to avoid a dynamic memroy allocation.
Also the `.c_str()` isn't needed since the AZ::Name function accepts a string_view.
The `.c_str()` is a pessimization, that causes a string_view to be constructed anyway and has to re-calculate strlen, where just passing in an AZStd::string/AZStd::fixed_string would construct the AZStd::string_view with their size member being slightly more performant.

```suggestion
                    const auto xrAttachmentName = AZStd::fixed_string<32>::format("XRSwapChain_View_%i", i);
                    xrDescriptor.m_attachmentId = RHI::AttachmentId{ xrAttachmentName };
```

Reply:
Thanks for the tip, I've never used `AZStd::fixed_string<32>::format` myself before. I'll give it a go.

I'll do the change locally and test it while I continue working, if everything works I'll upload this change in another PR.

Thanks.